package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Your second log doesn't show it, but I guess you get `Timeout running AT-command` there as well 15 seconds after sending the AT+CGDATA command. Since the failure happens after sending the interface up event, the connection stays up. Catching the CONNECT response in the runcommand.gcom should fix this.